### PR TITLE
[vi] Supervised Learning

### PR DESCRIPTION
**Pull request by tuananhhedspibk**

### CODE REVIEW

**Comment by tuananhhedspibk** [comment]
@shervinea I'm translating Supervised Learning Cheatsheet into Vietnamese, could you add `in progress` label for me? Thank you a lot.

File: vi/cheatsheet-supervised-learning.md [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+
+**8. [Discriminative model, Generative model, Goal, What's learned, Illustration, Examples]**
+
+&#10230; [Mô hình phân biệt, Mô hình sáng tạo, Mục tiêu, Những gì học được, Hình minh hoạ, Các ví dụ]
+
+<br>
+
+**9. [Directly estimate P(y|x), Estimate P(x|y) to then deduce P(y|x), Decision boundary,  	Probability distributions of the data, Regressions, SVMs, GDA, Naive Bayes]**
+
+&#10230; []
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; **[Uớc lượng trực tiếp P(x|y), ước lượng P(x|y) để tiếp tục suy luận P(x|y), Biên quyết định, Phân bố xác suất của dữ liệu, Hồi quy, SVMs, GDA, Naive Bayes]**
```

File: vi/cheatsheet-supervised-learning.md [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+
+**10. Notations and general concepts**
+
+&#10230; Kí hiệu và các khái niệm tổng quát
+
+<br>
+
+**11. Hypothesis ― The hypothesis is noted hθ and is the model that we choose. For a given input data x(i) the model prediction output is hθ(x(i)).**
+
+&#10230; Hypothesis - Hypothesis được kí hiệu là h0, là một mô hình mà chúng ta chọn. Với dữ liệu đầu vào cho trước x(i), mô hình dự đoaans đầu ra là h0(x(i)).
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; Hypothesis - Hypothesis được kí hiệu là h0, là một mô hình mà chúng ta chọn. Với dữ liệu đầu vào cho trước x(i), mô hình dự đoán đầu ra là h0(x(i)).
```

File: vi/cheatsheet-supervised-learning.md [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+
+**17. Remark: Stochastic gradient descent (SGD) is updating the parameter based on each training example, and batch gradient descent is on a batch of training examples.**
+
+&#10230; Chú ý: Stochastic gradient descent (SGD) là việc cập nhật tham số dựa theo mỗi ví dụ huấn luyện, và batch gradient descent là dựa trên một lô (batch) các ví dụ huấn luyện.
+
+<br>
+
+**18. Likelihood ― The likelihood of a model L(θ) given parameters θ is used to find the optimal parameters θ through maximizing the likelihood. In practice, we use the log-likelihood ℓ(θ)=log(L(θ)) which is easier to optimize. We have:**
+
+&#10230;
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; **18. Likelihood ― The likelihood của một mô hình L(θ) với bộ tham số θ được sử dụng để tối ưu bộ tham số θ thông qua việc tối đa hóa likelihood. Trong thực tế, chúng ta sử dụng log-likelihood ℓ(θ)=log(L(θ)) để tối ưu dễ dàng hơn. Chúng ta có:**
```

File: vi/cheatsheet-supervised-learning.md [hunk 4]
```diff
@@ -0,0 +1,567 @@
...
+
+**23. We assume here that y|x;θ∼N(μ,σ2)**
+
+&#10230; Chúng ta giả sử ở đây rằng y|x;θ∼N(μ,σ2)
+
+<br>
+
+**24. Normal equations ― By noting X the matrix design, the value of θ that minimizes the cost function is a closed-form solution such that:**
+
+&#10230; Phương trình normal - Bằng việc kí hiệu X là ma trận thiết kế, giá trị của θ mà cực tiểu hoá cost function là một phương pháp dạng đóng như là:
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; Phương trình chuẩn - Bằng việc kí hiệu X là ma trận thiết kế, giá trị của θ mà cực tiểu hoá cost function là một phương pháp dạng đóng như là:
```

File: vi/cheatsheet-supervised-learning.md [hunk 5]
```diff
@@ -0,0 +1,567 @@
...
+
+**50. [Non-linear separability, Use of a kernel mapping, Decision boundary in the original space]**
+
+&#10230; [Phân tách phi tuyến tính, Sử dụng kernel mapping, Ranh giới quyết định trong không gian gốc]
+
+<br>
+
+**51. Remark: we say that we use the "kernel trick" to compute the cost function using the kernel because we actually don't need to know the explicit mapping ϕ, which is often very complicated. Instead, only the values K(x,z) are needed.**
+
+&#10230;
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; **51. Chú ý: chúng ta nói rằng chúng ta sử dụng "kernel trick" để tính toán cost function  sử dụng kernel bởi vì chúng ta thực tế không cần hiểu cụ thể sự ánh xạ ϕ, những thứ thường rất phức tạp. Thay vào đó, chỉ các giá trị K(x,z) là cần thiết.**
```

File: vi/cheatsheet-supervised-learning.md [hunk 6]
```diff
@@ -0,0 +1,567 @@
...
+
+**52. Lagrangian ― We define the Lagrangian L(w,b) as follows:**
+
+&#10230; Lagrangian - Chúng ta định nghĩa Lagrangian L(w,b) như sau:
+
+<br>
+
+**53. Remark: the coefficients βi are called the Lagrange multipliers.**
+
+&#10230;
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; **53. Chú ý: Hệ số βi được gọi là bội số Lagrange.**
```

File: vi/cheatsheet-supervised-learning.md [hunk 7]
```diff
@@ -0,0 +1,567 @@
...
+
+**64. These methods can be used for both regression and classification problems.**
+
+&#10230;
+
+<br>
+
+**65. CART ― Classification and Regression Trees (CART), commonly known as decision trees, can be represented as binary trees. They have the advantage to be very interpretable.**
+
+&#10230;
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; CART - Cây phân loại và hồi quy (CART), được biết với tên phổ biến là cây quyết định, có thể được biểu diễn dưới dạng cây nhị phân. Chúng có các ưu điểm có thể diễn giả được.
```

File: vi/cheatsheet-supervised-learning.md [hunk 8]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**66. Random forest ― It is a tree-based technique that uses a high number of decision trees built out of randomly selected sets of features. Contrary to the simple decision tree, it is highly uninterpretable but its generally good performance makes it a popular algorithm.**
+
+&#10230;
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; Rừng ngẫu nhiên - Đây là một kĩ thuật dựa trên cây với việc sử dụng số lượng nhiều các cây cây quyết định để lựa chọn ngẫu nhiên các tập đặc trưng. Trái ngược với cây quyết định đơn giản, chúng khó có thể diễn giải nhưng vì hiệu năng cao nên trở thành thuật toán phổ biến.
```

File: vi/cheatsheet-supervised-learning.md [hunk 9]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**67. Remark: random forests are a type of ensemble methods.**
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
**67. Chú ý: rừng ngẫu nhiên là một loại giải thuật ensemble.**
```

File: vi/cheatsheet-supervised-learning.md [hunk 10]
```diff
@@ -0,0 +1,567 @@
...
+
+&#10230;
+
+<br>
+
+**68. Boosting ― The idea of boosting methods is to combine several weak learners to form a stronger one. The main ones are summed up in the table below:**
+
+&#10230;
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230;  Boosting ―  Ý tưởng của phương pháp boosting là kết hợp các phương pháp học yếu hơn để tạo nên một phương pháp học mạnh hơn. Các ý tưởng chính được tổng hợp ở bảng dưới:
```

File: vi/cheatsheet-supervised-learning.md [hunk 11]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**69. [Adaptive boosting, Gradient boosting]**
+
+&#10230;
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; Adaptive boosting, Gradient boosting
```

File: vi/cheatsheet-supervised-learning.md [hunk 12]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**70. High weights are put on errors to improve at the next boosting step**
+
+&#10230;
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]:
```suggestion
&#10230; Các trọng số cao được đặt vào các phần lỗi để cải thiện bước boosting tiếp theo
```

**Comment by tuananhhedspibk** [comment]
Hi @shervinea . I have force pushed to my brach `vi_translating_cheatsheet_supervised_learning`. Is there any problem if you merge this pull request to your project? Just tell me if you feel uncomfortable with my **force pushed commit**, I'll re-create a new pull request for this translation. Many thanks.

File: vi/cheatsheet-supervised-learning.md [hunk 13]
```diff
@@ -0,0 +1,567 @@
...
+
+**38. Assumptions of GLMs ― Generalized Linear Models (GLM) aim at predicting a random variable y as a function fo x∈Rn+1 and rely on the following 3 assumptions:**
+
+&#10230; Giả thuyết GLMs - Mô hình tuyến tính tổng quát (GLM) với mục đích là dự đoán một biến ngẫu nhiên y như là hàm cho biến x∈Rn+1 và dựa trên 3 giả thuyết sau:
+
+<br>
+
+**39. Remark: ordinary least squares and logistic regression are special cases of generalized linear models.**
+
+&#10230; Chú ý: Bình phương nhỏ nhất thông thường và logistirc regression đều là các trường hợp đặc biệt của các mô hình tuyến tính tổng quát.
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted.

File: vi/cheatsheet-supervised-learning.md [hunk 14]
```diff
@@ -0,0 +1,567 @@
...
+
+**50. [Non-linear separability, Use of a kernel mapping, Decision boundary in the original space]**
+
+&#10230; [Phân tách phi tuyến, Việc sử dụng một kernel mapping, Biến quyết định trong không gian gốc]
+
+<br>
+
+**51. Remark: we say that we use the "kernel trick" to compute the cost function using the kernel because we actually don't need to know the explicit mapping ϕ, which is often very complicated. Instead, only the values K(x,z) are needed.**
+
+&#10230; Chú ý: chúng ta nói rằng chúng ta sử dụng "kernel trick" để tính toán cost function sử dụng kernel bởi vì chúng ta thực sự không cần biết đến mapping tường minh ϕ, nó thường khá phức tạp. Thay vào đó, chỉ cần biết giá trị K(x,z).
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md [hunk 15]
```diff
@@ -0,0 +1,567 @@
...
+
+**54. Generative Learning**
+
+&#10230; Generative Learning
+
+<br>
+
+**55. A generative model first tries to learn how the data is generated by estimating P(x|y), which we can then use to estimate P(y|x) by using Bayes' rule.**
+
+&#10230; Một generative model đầu tiên cố gắng học cách dữ liệu được sinh ra thông qua việc ước lượng P(x|y), chúng ta có thể sau đó sử dụng P(x|y) để ước lượng P(y|x) bằng cách sử dụng luật Bayes.
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md [hunk 16]
```diff
@@ -0,0 +1,567 @@
...
+
+**57. Setting ― The Gaussian Discriminant Analysis assumes that y and x|y=0 and x|y=1 are such that:**
+
+&#10230; Thiết lập - Gaussian Discriminant Analysis giả sử rằng y và x|y=0 và x|y=1 là như sau:
+
+<br>
+
+**58. Estimation ― The following table sums up the estimates that we find when maximizing the likelihood:**
+
+&#10230;
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md [hunk 17]
```diff
@@ -0,0 +1,567 @@
...
+
+**59. Naive Bayes**
+
+&#10230; Naive Bayes
+
+<br>
+
+**60. Assumption ― The Naive Bayes model supposes that the features of each data point are all independent:**
+
+&#10230;
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md [hunk 18]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**61. Solutions ― Maximizing the log-likelihood gives the following solutions, with k∈{0,1},l∈[[1,L]]**
+
+&#10230;
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md [hunk 19]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**62. Remark: Naive Bayes is widely used for text classification and spam detection.**
+
+&#10230;
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md [hunk 20]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**63. Tree-based and ensemble methods**
+
+&#10230; Phương thức Tree-based và toàn thể
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md [hunk 21]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**64. These methods can be used for both regression and classification problems.**
+
+&#10230;
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md (hunk 7 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md (hunk 8 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md (hunk 9 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md (hunk 10 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md (hunk 11 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

File: vi/cheatsheet-supervised-learning.md (hunk 12 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted

**Comment by tuananhhedspibk** [comment]
Hi @damminhtien . Many thanks to your review, I have finished my translation on **cheatsheet_supervised_learning**. Could you continue reviewing its content for me?

**Comment by tuananhhedspibk** [comment]
@shervinea I have finished my translation on **cheatsheet_supervised_learning**. Could you change the label of this pull request for me? Thanks.

**Comment by damminhtien** [comment]
> Hi @damminhtien . Many thanks to your review, I have finished my translation on **cheatsheet_supervised_learning**. Could you continue reviewing its content for me?

In my view, your work is totally awesome. No longer comment. Let's others review.

**Comment by shervinea** [comment]
@tuananhhedspibk Thank you for all your work! Just did it.

File: vi/cheatsheet-supervised-learning.md [hunk 22]
```diff
@@ -0,0 +1,567 @@
...
+
+**8. [Discriminative model, Generative model, Goal, What's learned, Illustration, Examples]**
+
+&#10230; [Mô hình phân biệt, Mô hình sinh, Mục tiêu, Những gì học được, Hình minh hoạ, Các ví dụ]
+
+<br>
+
+**9. [Directly estimate P(y|x), Estimate P(x|y) to then deduce P(y|x), Decision boundary,  	Probability distributions of the data, Regressions, SVMs, GDA, Naive Bayes]**
+
+&#10230; [Ước lượng trực tiếp P(y|x), Ước lượng P(x|y) để tiếp tục suy luận P(y|x), Biên quyết địnhđịnh, Phân bố xác suất của dữ liệu, Hồi quy, SVMs, GDA, Naive Bayes]
```
Comment (minhbk) [inline vi/cheatsheet-supervised-learning.md]: Biên quyết địnhđịnh

File: vi/cheatsheet-supervised-learning.md [hunk 23]
```diff
@@ -0,0 +1,567 @@
...
+
+**62. Remark: Naive Bayes is widely used for text classification and spam detection.**
+
+&#10230; Chú ý: Naive Bayes được sử dụng rộng rãi cho bài toán phân loại văn bản và phát hiện spam.
+
+<br>
+
+**63. Tree-based and ensemble methods**
+
+&#10230; Phương thức Tree-based và toàn thể
```
Comment (hungnguyen95) [inline vi/cheatsheet-supervised-learning.md]: ensemble nên để nguyên, không nên dịch ra tiếng việt

Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted.

File: vi/cheatsheet-supervised-learning.md [hunk 24]
```diff
@@ -0,0 +1,567 @@
...
+
+**5. [Regression, Classifier, Outcome, Examples]**
+
+&#10230; [Hồi quy, Phân loại, Đầu ra, Các ví dụ]
+
+<br>
+
+**6. [Continuous, Class, Linear regression, Logistic regression, SVM, Naive Bayes]**
+
+&#10230; [Tiếp diễn, Lớp, Hồi quy tuyến tính, Hồi quy Logistic, SVM, Naive Bayes]
```
[the remark this answers is not in the document]
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted.

File: vi/cheatsheet-supervised-learning.md (hunk 22 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: Review accepted.

**Comment by tuananhhedspibk** [comment]
Hey @minhbk , @damminhtien , could you guys check this translation 's content for me once more times? Many thanks!

File: vi/cheatsheet-supervised-learning.md [hunk 25]
```diff
@@ -0,0 +1,567 @@
...
+
+**38. Assumptions of GLMs ― Generalized Linear Models (GLM) aim at predicting a random variable y as a function fo x∈Rn+1 and rely on the following 3 assumptions:**
+
+&#10230; Giả thuyết GLMs - Mô hình tuyến tính tổng quát (GLM) với mục đích là dự đoán một biến ngẫu nhiên y như là hàm cho biến x∈Rn+1 và dựa trên 3 giả thuyết sau:
+
+<br>
+
+**39. Remark: ordinary least squares and logistic regression are special cases of generalized linear models.**
+
+&#10230; Chú ý: Bình phương nhỏ nhất thông thường và logistic regression đều là các trường hợp đặc biệt của các mô hình tuyến tính tổng quát.
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]: I suggest to replace 'regression' to 'hồi quy' to synchronization with your above translation.
```suggestion
&#10230; Chú ý: Bình phương nhỏ nhất thông thường và hồi quy logistic đều là các trường hợp đặc biệt của các mô hình tuyến tính tổng quát.
```

File: vi/cheatsheet-supervised-learning.md [hunk 26]
```diff
@@ -0,0 +1,567 @@
...
+
+**73. k-nearest neighbors ― The k-nearest neighbors algorithm, commonly known as k-NN, is a non-parametric approach where the response of a data point is determined by the nature of its k neighbors from the training set. It can be used in both classification and regression settings.**
+
+&#10230; k-nearest neighbors - Giải thuật k-nearest neighbors, thường được biết đến là k-NN, là cách tiếp cận phi-tham số, ở phương pháp này phân lớp của một điểm dữ liệu được định nghĩa bởi k điểm dữ liệu gần nó nhất trong tập huấn luyện. Phương pháp này có thể được sử dụng trong quá trình thiết lập cho bài toán phân loại cũng như bài toán hồi quy.
+
+<br>
+
+**74. Remark: The higher the parameter k, the higher the bias, and the lower the parameter k, the higher the variance.**
+
+&#10230; Chú ý: Tham số k cao hơn, bias cao hơn, tham số k thấp hơn, phương sai cao hơn
```
Comment (damminhtien) [inline vi/cheatsheet-supervised-learning.md]: You can translate 'bias' to 'độ chệch'. I found [AIVIVN](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation) was using it.

File: vi/cheatsheet-supervised-learning.md (hunk 25 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: review accepted

File: vi/cheatsheet-supervised-learning.md (hunk 26 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: review accepted

File: vi/cheatsheet-supervised-learning.md (hunk 25 above)
Comment (tuananhhedspibk) [reply vi/cheatsheet-supervised-learning.md]: review accepted

**Comment by tuananhhedspibk** [comment]
@damminhtien , I have fixed it. Please check once more time for me. Thanks

**Comment by shervinea** [comment]
Thank you @tuananhhedspibk, @damminhtien, @hungnguyen95 and @minhbk for all your hard work! Merging this PR right now.

**Pull request merged**